src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,101 @@
+package duke.parser;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.common.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+public abstract class Parser {
+
+    public static Command parse(String input) throws DukeException {
+        String[] commandArgs = input.trim().split(" ", 2);
+        String commandType = commandArgs[0];
+        Command command;
+        switch (commandType) {
```

Comment:
I like your use of switch flow control! Much better than using a whole bunch of if-else statements!